### ISSUE TITLE
Install mds-provider Client from Santa Monica

### ISSUE THREAD

**Issue by mateoclarke** [cityofaustin/atd-data-tech#1070]
- We originally forked from this repo but now we need to catch up with features in this client that support versioning, specifically for 0.4.0. https://github.com/CityofSantaMonica/mds-provider
- It probably makes sense to try to start an install from a clean slate instead of trying to update our fork but either way could work, I think.
- We also need to check which providers are supporting MDS 0.4.0 and if not, what is the roadmap to upgrade.

- [x] install mds-provider and begin supporting 0.4.0
- [x] check what version our providers are supporting.

**Comment by sergiogcx** [comment]
I've managed to install the santa monica mds-provider library and run it, but not without major problems.

I managed to access data from Bird, Lime, and Jump; however, I could not access data from Lyft, Spin, or Veoride.

Lyft: 
- The mds-provider library does not support basic auth, and it does not provide a way to add that capability or extend the code to make it authenticate differently. 
- I also tried passing encoded headers needed for authentication, but the library attempts to use Bearer authentication as opposed to Basic within the HTTP headers.
- It may be possible to check if Lyft has any other way of authentication. I meant to ask Kegan Maher (the current maintainer of the santa monica library) how they are accessing Lyft, but I haven't done so in the hopes I could fix this.

Spin:
- The mds-provider library is able to authenticate, and even while Spin claims to be 0.4 compatible, the library insists in using version 0.3.
- While Spin is also compatible with version 0.3, the endpoint returns an error message about incorrect http URI parameters. There is no way to change it within the configuration or the code.

One major problem the library has, is that it is very opinionated on how the clients should behave. It relies on its own database to find url endpoints, and it is difficult to troubleshoot what it is doing and read the code. There is no documentation at all or any examples on how to use different providers.

While testing and troubleshooting the library, I found myself able to use CURL or Postman to make successful requests, but given the narrow way the library works I just could not make the small adjustments in the settings. After spending a few days going through the code and understanding as well as trying to make the library work, I realized I could have implemented a decent library myself.

**Comment by sergiogcx** [comment]
With Santa Monica's client, I spent a couple days trying to make work and it still remained an enigma and most of the code is hard to read, much more difficult to maintain and I could not explain its behavior to the best of my efforts. Not only that, the library does not allow extending the code, the library itself needs to be updated in order to make provider-specific changes.

In three-four days, I wrote an library that is compatible with MDS 0.3.2, and has been packaged and ready to go:
https://github.com/cityofaustin/atd-mds-client

No providers (as far as I could tell) support MDS 0.4; bird and lime are the most up-to-date, they both support 0.3.2.

The new library was written with extensibility in a non-opinionated  approach, meaning that the code provides basic functionality for MDS 2 and 3, but allows the developer to extend the code and write new drivers and custom parameters.
